src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -182,6 +182,113 @@ public virtual void NextBytes(byte[] buffer)
+        ///   The method uses <see cref="Next(int)" /> to select items randomly from <paramref name="choices" />
+        ///   by index and populate <paramref name="destination" />.
+        /// </remarks>
+        public void GetItems<T>(ReadOnlySpan<T> choices, Span<T> destination)
+        {
+            if (choices.IsEmpty)
+                throw new ArgumentException(SR.Arg_EmptySpan, nameof(choices));
+
+            for (int i = 0; i < destination.Length; i++)
+            {
+                destination[i] = choices[Next(choices.Length)];
+            }
+        }
+
+        /// <summary>
+        ///   Creates an array populated with items chosen at random from choices.
+        /// </summary>
+        /// <param name="choices">The items to use to populate the array.</param>
+        /// <param name="length">The length of array to return.</param>
+        /// <typeparam name="T">The type of array.</typeparam>
+        /// <returns>An array populated with random items.</returns>
+        /// <exception cref="ArgumentException"><paramref name="choices" /> is empty.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="choices" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="length" /> is not zero or a positive number.
+        /// </exception>
+        /// <remarks>
+        ///   The method uses <see cref="Next(int)" /> to select items randomly from <paramref name="choices" />
+        ///   by index. This is used to populate a newly-created array.
+        /// </remarks>
+        public T[] GetItems<T>(T[] choices, int length)
+        {
+            ArgumentNullException.ThrowIfNull(choices);
+            return GetItems(new ReadOnlySpan<T>(choices), length);
+        }
+
+        /// <summary>
+        ///   Creates an array populated with items chosen at random from choices.
+        /// </summary>
+        /// <param name="choices">The items to use to populate the array.</param>
+        /// <param name="length">The length of array to return.</param>
+        /// <typeparam name="T">The type of array.</typeparam>
+        /// <returns>An array populated with random items.</returns>
+        /// <exception cref="ArgumentException"><paramref name="choices" /> is empty.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="length" /> is not zero or a positive number.
+        /// </exception>
+        /// <remarks>
+        ///   The method uses <see cref="Next(int)" /> to select items randomly from <paramref name="choices" />
+        ///   by index. This is used to populate a newly-created array.
+        /// </remarks>
+        public T[] GetItems<T>(ReadOnlySpan<T> choices, int length)
+        {
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            T[] items = new T[length];
```

Comment:
If we think there's any reasonable chance length could be 0, we should consider special-casing it:
```C#
if (length == 0)
{
    return Array.Empty<T>();
}
```

Reply:
I couldn't _really_ see a solid use case for it and it seemed like special casing a not-so-common path that the common path would be behind. This feels similar to this feedback previously: https://github.com/dotnet/runtime/pull/43221#discussion_r502571729

Reply:
Yup.  That's really my question, whether there's a real use case for it, e.g. do we expect the length is always hardcoded or might the length be computed with math such that 0 might happen.  I'm fine leaving it out for now and adding it later if it's noticed to be an issue.

Reply:
Okay. I'm inclined to leave it out for now (same for the one further down).